### ISSUE TITLE
ICSx⁵ crashes with a Security-Exception at at.bitfire.icsdroid.AppAccount.get

### ISSUE THREAD

**Issue by devvv4ever** [bitfireAT/icsx5#18]
It seems the system can't access the calendar list of an ICSx⁵ account (**at.bitfire.icsdroid.AppAccount.get**)? Maybe we should just try-catch this and handle this case properly?

This happens on many different **Samsung devices** so far as well as on a Google Pixel 5 and so far only on **Android 11 and 12**.

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:4022)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:4188)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:103)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2425)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:226)
  at android.os.Looper.loop (Looper.java:313)
  at android.app.ActivityThread.main (ActivityThread.java:8582)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:563)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1133)
Caused by: java.lang.SecurityException: 
  at android.os.Parcel.createExceptionOrNull (Parcel.java:2437)
  at android.os.Parcel.createException (Parcel.java:2421)
  at android.os.Parcel.readException (Parcel.java:2404)
  at android.os.Parcel.readException (Parcel.java:2346)
  at android.accounts.IAccountManager$Stub$Proxy.addAccountExplicitly (IAccountManager.java:1550)
  at android.accounts.AccountManager.addAccountExplicitly (AccountManager.java:1030)
  at at.bitfire.icsdroid.AppAccount.get (AppAccount.kt:50)
  at at.bitfire.icsdroid.ui.CalendarListActivity$CalendarModel.loadCalendars (CalendarListActivity.kt:332)
  at at.bitfire.icsdroid.ui.CalendarListActivity$CalendarModel.startWatchingCalendars (CalendarListActivity.kt:318)
  at at.bitfire.icsdroid.ui.CalendarListActivity$CalendarModel.reinit (CalendarListActivity.kt:301)
  at at.bitfire.icsdroid.ui.CalendarListActivity.onCreate (CalendarListActivity.kt:98)
  at android.app.Activity.performCreate (Activity.java:8282)
  at android.app.Activity.performCreate (Activity.java:8262)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1329)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3996)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:4188)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:103)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:135)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:95)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2425)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:226)
  at android.os.Looper.loop (Looper.java:313)
  at android.app.ActivityThread.main (ActivityThread.java:8582)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:563)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1133)
Caused by: android.os.RemoteException: 
  at com.android.server.accounts.AccountManagerService.addAccountExplicitlyWithVisibility (AccountManagerService.java:501)
  at com.android.server.accounts.AccountManagerService.addAccountExplicitly (AccountManagerService.java:1776)
  at android.accounts.IAccountManager$Stub.onTransact (IAccountManager.java:583)
  at com.android.server.accounts.AccountManagerService.onTransact (AccountManagerService.java:1119)
  at android.os.Binder.execTransactInternal (Binder.java:1215)
```

**Comment by rfc2822** [comment]
I wonder why it should ever be prohibited to add an account of an own account type (= the type where we have an authenticator.)

Do you have an idea @ArnyminerZ?

**Comment by ArnyminerZ** [comment]
I've been researching about this for a long time, and I have not been able to find a reason. All the errors of this kind are forgotten or miswritten authenticators. So... If it only happened on Samsung devices I would be blaming OneUI, but that Pixel breaks me. We could report the issue to Google, and see what they have to say, but I don't think it would be really useful if we are not able to reproduce it.

**Comment by ArnyminerZ** [comment]
I have however found a short thread on [Samsung's dev forum](https://forum.developer.samsung.com/t/securityexception-with-accountmanager-service/16112) that seems to have the same issue. This leads to an [issue on Google](https://issuetracker.google.com/issues/192000156) which talk about Samsung devices on Android 11 and 12, but there's no answer from Google's side.

**Comment by rfc2822** [comment]
Still happens with v2.0.3. Let's see whether it still occurs with v2.1, but I guess so since nothing has changed in terms of account creation.

There are a few things that could be checked: https://stackoverflow.com/questions/33270613/android-app-crashes-on-addaccountexplicitlyaccount-password-null

**Comment by rfc2822** [comment]
@ArnyminerZ Do you think one of the things mentioned in the StackOverflow link above could apply to our case?

**Comment by ArnyminerZ** [comment]
I did take a look at them, and they doesn't seem to fix the issue. However, a week ago or so I was having some issues with multiple accounts in Android. I can't have it reproduced, but when adding two accounts to the system, and them removing one, it seems like there's a bug in Android that gives the app access to an account which has not been created with the app.

Even though I'm listing accounts with the correct account type, Android returns some that don't have this type, super strange, and not consistent.

There must be something wrong on Android's side, maybe I can take a look at Google's issue tracker, and see if there's something reported, otherwise try to reproduce and report it...

**Comment by rfc2822** [comment]
Seems to be fixed with 2.1, because the subscription list doesn't create the account directly anymore.
